src/promptflow/promptflow/_sdk/_start_trace.py

```diff
@@ -50,34 +46,23 @@ def start_trace():
 
 def _start_pfs_in_background(pfs_port) -> None:
     """Start a pfs process in background."""
+    args = [sys.executable, "-m", "promptflow._sdk._service.entry", "start", "--port", str(pfs_port)]
     if is_port_in_use(pfs_port):
-        _logger.debug("PFS is already serving on port %s", pfs_port)
-        return
-    args = [sys.executable, "-m", "promptflow._sdk._service.entry", "start"]
+        _logger.warning(f"Service port {pfs_port} is used.")
```

Comment:
we shoud have this logic extracted as function also

Reply:
extracted